### ISSUE TITLE
Wake up on connect

### ISSUE THREAD

**Issue by SebastianPfliegel** [thestr4ng3r/chiaki#42]
When trying to connect to a console in standby the attempt fails. It'd be nice to fire always a wake up call to the console prior to the connection attempt.

**Comment by bubelov** [comment]
Same on my laptop.

OS: **Ubuntu 19.04**
App version: **1.0.3**
Console: **PS4 Pro**

![Screenshot from 2019-09-29 13-13-35](https://user-images.githubusercontent.com/1189245/65827609-0eb54300-e2bc-11e9-97b7-72dc5f9d8dbe.png)

![Screenshot from 2019-09-29 13-14-51](https://user-images.githubusercontent.com/1189245/65827615-1ecd2280-e2bc-11e9-930e-64e72f3e7e50.png)

After pressing "Yes", nothing seems to happen. I didn't notice any related logs in both default and verbose modes.

Fantastic app, by the way!

**Comment by SebastianPfliegel** [comment]
I don't even get that prompt, @bubelov

**Comment by thestr4ng3r** [comment]
The prompt is only shown for non-manually added consoles when it is known that the console is in standby mode.

For waking over the internet, you would have to forward port 987 (UDP), however I would discourage you from doing so because the protocol Sony uses for that allows for trivial replay attacks. This is also the reason why I will not send wakeup packets without the user's explicit approval.

If you are on a local network and wakeup fails, you can check with Wireshark or tcpdump on port 987 whether the packet is sent. (If you do, keep the raw packet data secret because of the security issue mentioned above)

**Comment by bubelov** [comment]
@thestr4ng3r it's a local network. Here is the output of tcpdump:

```bash
IP xps.lan.34318 > 255.255.255.255.987: UDP, length 60
```

It seems to send it multiple times to no avail.

**Comment by aFoxNamedMorris** [comment]
> @thestr4ng3r it's a local network. Here is the output of tcpdump:
> 
> ```shell
> IP xps.lan.34318 > 255.255.255.255.987: UDP, length 60
> ```
> 
> It seems to send it multiple times to no avail.

Did you make sure to allow wake on LAN in your PS4's power settings?

**Comment by bubelov** [comment]
@aFoxNamedMorris no, I didn't see any instructions about that. I've found that setting and enabled it and it did the trick, thanks for your hint!

```bash
[I] Discovery Service detected new host with id xxx
[I] Discovery Service: Host with id xxx is no longer available
[I] Discovery Service detected new host with id xxx
```

The console have disappeared for a couple of minutes and then came back with the new sate (ready). I've tried to open a session and it worked as usual. This disappearance thing was unexpected (no warning, no progress bar) but I guess those are known low priority issues.

It would be nice to have set up instructions added to a project readme and probably to the app itself.

**Comment by Rodrigo-Barros** [comment]
> > @thestr4ng3r it's a local network. Here is the output of tcpdump:
> > ```shell
> > IP xps.lan.34318 > 255.255.255.255.987: UDP, length 60
> > ```
> > 
> > 
> > It seems to send it multiple times to no avail.
> 
> Did you make sure to allow wake on LAN in your PS4's power settings?

Thanks I wondering how to make this feature work, very helpful your response.
